csv.tpl

```diff
@@ -1,53 +1,98 @@
-	{{- if (eq (len .Title) 0) }}
-		{{- printf "%s: %s - %s severity vulnerability" .PkgName .InstalledVersion .Vulnerability.Severity | escapeCsv }}
-	{{- else }}
-		{{- abbrev 100 .Title | escapeCsv }}
-	{{- end }},
-	{{- abbrev 500 .Vulnerability.Description | escapeCsv }},
-	{{- if .FixedVersion }}
-		{{- printf "Update %s to version %s or higher." .PkgName .FixedVersion | escapeCsv }}
-	{{- else }}
-		{{- printf "No resolution provided." | escapeCsv }}
-	{{- end }},
-	{{- .PrimaryURL | escapeCsv }},
-	{{- $reference := false }}
-	{{- range .References }}
-		{{- if contains "nvd.nist.gov" . }}
-			{{- . | escapeCsv }}
-			{{- $reference = true }}
-		{{- end }}
+	{{- if isFieldAvailable "Target" }}
+	    {{- $target | escapeCsv }},
 	{{- end }}
-	{{- if not $reference }}
-		{{- printf "" | escapeCsv }}
-	{{- end }},
+	{{- if isFieldAvailable "Vulnerability Class" }}
+	    {{- printf "%s" $class | escapeCsv }},
+    {{- else -}}
+	{{- end }}
+	{{- if isFieldAvailable "Target Type" }}
+	    {{- $vulnerabilityType | escapeCsv }},
+    {{- else -}}
+	{{- end }}
+	{{- if isFieldAvailable "Vulnerability ID" }}
+	    {{- .VulnerabilityID | escapeCsv }},
+    {{- else -}}
+	{{- end }}
+	{{- if isFieldAvailable "Severity" }}
+	    {{- .Vulnerability.Severity | escapeCsv }},
+    {{- else -}}
+	{{- end }}
+	{{- if isFieldAvailable "PackageName" }}
+	    {{- .PkgName | escapeCsv }},
+    {{- else -}}
+	{{- end }}
+	{{- if isFieldAvailable "Installed Version" }}
+	    {{- .InstalledVersion | escapeCsv }},
+    {{- else -}}
+	{{- end }}
+	{{- if isFieldAvailable "Fixed Version" }}
+        {{- .FixedVersion | escapeCsv }},
+    {{- else -}}
+	{{- end }}
+	{{- if isFieldAvailable "Title" }}
+        {{- if (eq (len .Title) 0) }}
+            {{- printf "%s: %s - %s severity vulnerability" .PkgName .InstalledVersion .Vulnerability.Severity | escapeCsv }},
+        {{- else }}
+            {{- abbrev 100 .Title | escapeCsv }},
+        {{- end }}
+    {{- else -}}
+	{{- end }}
+	{{- if isFieldAvailable "Description" }}
+	    {{- abbrev 500 .Vulnerability.Description | escapeCsv }},
+    {{- else -}}
+	{{- end }}
+	{{- if isFieldAvailable "Resolution" }}
+        {{- if .FixedVersion }}
+            {{- printf "Update %s to version %s or higher." .PkgName .FixedVersion | escapeCsv }},
+        {{- else }}
+            {{- printf "No resolution provided." | escapeCsv }},
+        {{- end }}
+    {{- else -}}
+	{{- end }}
+	{{- if isFieldAvailable "Reference" }}
+	    {{- .PrimaryURL | escapeCsv }},
+    {{- else -}}
+	{{- end }}
+	{{- if isFieldAvailable "Additional Reference" }}
+        {{- $reference := false }}
+        {{- range .References }}
+            {{- if contains "nvd.nist.gov" . }}
+                {{- . | escapeCsv }}
+                {{- $reference = true }}
+            {{- end }}
+        {{- end }}
+        {{- if not $reference }}
+            {{- printf "" | escapeCsv }}
+        {{- end }},
+    {{- else -}}
+	{{- end }}
+
```

Comment:
it seems this template was copied from Trivy.
`csv-report` should implement the single feature.
so we can move any logic to `main.go`, the template will save the table only. 
maybe simple formatting only like `printf "%.1f"`